src/support.rs

```diff
@@ -506,6 +509,46 @@ impl<F> FieldOffset<F> {
   }
 }
 
+/// A special hasher that is optimized for hashing `std::any::TypeId` values.
+/// It can't be used for anything else.
+#[derive(Clone, Default)]
+pub(crate) struct TypeIdHasher {
+  state: Option<u64>,
+}
```

Comment:
Seems unnecessary but sure...

Reply:
Not super necessary I suppose but more like a nice thing.